### PR TITLE
Respect M2 tax display setting to show prices with taxes

### LINKED ISSUE #1364: Respect M2 tax display setting to show prices with taxes
**Description**:
Could we show product prices with taxes in shopping cart also? 
Even if I set:
Stores-- Configuration -- Sales -- Tax -- Shopping Cart Display Settings --- All values to Including Tax
still show with out tax

**Additional context**:
I think this can be confusing to normal customers (who buy with taxes)

### PR DESCRIPTION
**Pull request by zans-laksa**
Fixes #1364
Related pull request: scandipwa/quote-graphql/pull/45

### CODE REVIEW

File: src/app/component/CartItemPrice/CartItemPrice.container.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        if (displayTaxInPrice === DISPLAY_CART_TAX_IN_PRICE_BOTH) {
+            return row_total;
+        }
+
+        return null;
+    }
+
+    render() {
+        return <CartItemPrice { ...this.props } { ...this.containerProps() } />;
```
Comment (yeegor) [inline src/app/component/CartItemPrice/CartItemPrice.container.js:83]: Please make this multiline, see other elements in the application as a reference.

File: src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 2]
```diff
@@ -132,10 +140,89 @@ export class CheckoutOrderSummary extends PureComponent {
...
+            }
+        } = this.props;
+
+        if (!price) {
+            return null;
+        }
+
+        return (
+            <span>
+                { `${ __('Excl. tax:') } ${ formatPrice(price || 0, quote_currency_code) }` }
```
Comment (yeegor) [inline src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:156]: Please convert `price || 0` to default function's argument value

File: src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 3]
```diff
@@ -56,7 +62,9 @@ export class CheckoutOrderSummary extends PureComponent {
                 </strong>
                 <strong block="CheckoutOrderSummary" elem="Text">
                     { `${priceString}` }
+                    { subPrice }
```
Comment (yeegor) [inline src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:65]: Consider using `renderThing` pattern here and determine whether to render this or not depending on the value you process there. Too lot of args for one function :)

**Review by yeegor — CHANGES_REQUESTED** [review]
Minor coding style comments

File: src/app/component/CartItemPrice/CartItemPrice.container.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+/** @namespace Component/CartItemPrice/Container/mapStateToProps */
+export const mapStateToProps = (state) => {
+    const {
+        CartReducer: {
+            cartTotals: {
+                cart_display_config: {
+                    display_tax_in_price
+                } = {}
+            }
```
[the remark this answers is not in the document]
Comment (zans-laksa) [reply src/app/component/CartItemPrice/CartItemPrice.container.js:30]: Yes, it's always present.

File: src/app/component/CartItemPrice/CartItemPrice.container.js (hunk 1 above)
Comment (zans-laksa) [reply src/app/component/CartItemPrice/CartItemPrice.container.js]: Done

File: src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js (hunk 2 above)
Comment (zans-laksa) [reply src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js]: Done

File: src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js (hunk 3 above)
Comment (zans-laksa) [reply src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js]: There are multiple calls of this helper function with the very different args.
Moved it to a separate component.

File: src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 5]
```diff
@@ -144,18 +215,101 @@ export class CheckoutOrderSummary extends PureComponent {
...
-                        : null }
+                    { this.renderSubTotal() }
+                    { this.renderShipping() }
                     { this.renderDiscount() }
-                    { this.renderPriceLine(tax_amount, __('Tax')) }
-                    { checkoutStep !== SHIPPING_STEP
-                        ? this.renderPriceLine(grand_total + shipping_amount, __('Order total'))
-                        : this.renderPriceLine(subtotal_with_discount + tax_amount, __('Order total')) }
+                    { this.renderTax() }
+                    { this.renderOrderTotal() }
```
Comment (yeegor) [inline src/app/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:312]: This looks much better after your modifications! Good job!

**Review by yeegor — APPROVED** [review]
Good.

**Pull request merged**